### ISSUE TITLE
reportUnnecessaryCast doesn't work for literals

### ISSUE THREAD

**Issue by parched** [microsoft/pyright#4806]
**Screenshots or Code**
```python
x = cast(int, 0) # expected reportUnnecessaryCast
```

**VS Code extension or command-line**
Commandline 1.1.299

**Comment by erictraut** [comment]
This isn't an unnecessary cast. It's casting `Literal[0]` to `int`, so it has an effect. The `reportUnnecessaryCast` diagnostic reports only those cases where the cast has no effect and is therefore unnecessary.

**Comment by parched** [comment]
Interesting, I guess this is to force up-casting, but using `cast` to do up-casting seems very unsafe. Perhaps there needs to be a `typing.up_cast` function.

**Comment by erictraut** [comment]
There has been some discussion of this (a "safe cast" call) in the typing forums, but nothing has come of it yet.
